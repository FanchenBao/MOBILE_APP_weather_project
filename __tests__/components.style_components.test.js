import React from 'react';
import renderer from 'react-test-renderer';
import {
  CityText,
  ErrorText,
  ForecastCategoryText,
  ForecastContentText,
  ErrorBubble,
} from '../src/components/style_components.js';

it('renders CityText correctly', () => {
  const tree = renderer.create(<CityText>Test for CityText</CityText>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ErrorText correctly', () => {
  const tree = renderer
    .create(<ErrorText>Test for ErrorText</ErrorText>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ForecastCategoryText correctly', () => {
  const tree = renderer
    .create(
      <ForecastCategoryText>
        Test for ForecastCategoryText
      </ForecastCategoryText>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ForecastContentText correctly', () => {
  const tree = renderer
    .create(
      <ForecastContentText>Test for ForecastContentText</ForecastContentText>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ErrorBubble correctly', () => {
  const tree = renderer
    .create(<ErrorBubble>Test for ErrorBubble</ErrorBubble>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
