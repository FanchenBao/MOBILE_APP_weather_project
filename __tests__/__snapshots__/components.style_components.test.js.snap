// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CityText correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 40,
      },
      Object {
        "color": "#66ff33",
      },
    ]
  }
>
  Test for CityText
</Text>
`;

exports[`renders ErrorBubble correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "backgroundColor": "red",
        "borderRadius": 10,
        "color": "white",
        "marginHorizontal": 10,
        "marginTop": 5,
        "padding": 2,
        "textAlign": "center",
      },
      Object {
        "fontSize": 14,
      },
    ]
  }
>
  Test for ErrorBubble
</Text>
`;

exports[`renders ErrorText correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 40,
      },
      Object {
        "backgroundColor": "red",
        "borderRadius": 10,
        "color": "white",
        "fontSize": 20,
        "padding": 10,
        "textAlign": "center",
      },
    ]
  }
>
  Test for ErrorText
</Text>
`;

exports[`renders ForecastCategoryText correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 20,
      },
      Object {
        "fontWeight": "bold",
      },
      Object {
        "marginLeft": 15,
      },
      Object {
        "color": "#66ff33",
      },
    ]
  }
>
  Test for ForecastCategoryText
</Text>
`;

exports[`renders ForecastContentText correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontSize": 20,
      },
      Object {
        "marginRight": 30,
      },
      Object {
        "color": "#66ff33",
      },
    ]
  }
>
  Test for ForecastContentText
</Text>
`;
